opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -1028,8 +1028,37 @@ def make_model_config_json(
         if model_name is None:
             model_name = self.model_id
 
-        # if user input model_type and embedding_dimension, it will skip reading the config.json file
-        if model_type is None or embedding_dimension is None:
+        # if user input model_type/embedding_dimension/pooling_mode, it will skip this step.
+
+        model = SentenceTransformer(self.model_id)
+        if model_type is None:
+            if len(model._modules) >= 1 and isinstance(
+                model._modules["0"], Transformer
+            ):
+                try:
+                    model_type = model._modules["0"].auto_model.__class__.__name__
+                    model_type = model_type.lower().rstrip("model")
+                except Exception as e:
+                    raise Exception(f"Raised exception while getting model_type: {e}")
```

Comment:
Let's try to cover these exception branches. I think either we can mock the model or we can stub the `_modules` as None so that it can go to the except branch. Same for other exceptions.

Reply:
I don't think we can do it that way because the function loads a hugging face SentenceTransformer within the function, so we cannot stub the _modules as None in test file.

Reply:
I agree this will be bit tricky to implement. We can skip this for now. Not urgent.